src/test/run-make/reproducible-build/Makefile

```diff
@@ -10,31 +10,27 @@ all:  \
 
 smoke:
 	rm -rf $(TMPDIR) && mkdir $(TMPDIR)
+	$(RUSTC) linker.rs -O
```

Comment:
I rewrote this test to be a little more resilient against system linkers. We're sort of always assuming the system linker is deterministic, and if it's not then there's not much we can do about that...